packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js

```diff
@@ -11,20 +11,33 @@
 const chalk = require('chalk');
 const fs = require('fs');
 const resolve = require('resolve');
+const path = require('path');
 const paths = require('../../config/paths');
 
 function verifyTypeScriptSetup() {
-  if (!fs.existsSync(paths.appTsConfig)) {
+  if (!paths.appIndexJs.endsWith('.ts') || !paths.appIndexJs.endsWith('.tsx')) {
```

Comment:
Shouldn't this be &&? 

Also just fyi that the index file is not required to be .ts or .tsx on typescript projects.